classes/ProductSale.php

```diff
@@ -82,19 +82,22 @@ public static function getBestSales($idLang, $pageNumber = 0, $nbProducts = 10,
         $finalOrderBy = $orderBy;
         $orderTable = '';
 
-        $invalidOrderBy = !Validate::isOrderBy($orderBy);
-        if ($invalidOrderBy || null === $orderBy) {
-            $orderBy = 'quantity';
-            $orderTable = 'ps';
-        }
-
-        if ($orderBy == 'date_add' || $orderBy == 'date_upd') {
-            $orderTable = 'product_shop';
-        }
+        if (strtolower($orderWay) === 'random') {
+            $orderWay = 'RAND()';
```

Comment:
Well I did not know you could do this 😄  . It's quite powerful.

Reply:
Hello @matks 

I understand what you mean but I don't see why anyone would do a random if the query can't accept it.

`random` is also introduced in : 

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Product/Search/SortOrder.php#L202-L212

For me it's a heresy to randomize the best sellers but an issue has been opened for that.

I want to change things but I don't see how?  Do you have a way to help me?

Reply:
@matks You didn't know the `ORDER BY RAND()`? :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream: :scream:

Reply:
@PierreRambaud No, I did not know.

Do I even know SQL ?

Reply:
![Heart attack](https://giphy.com/embed/f7GXfUvpLXNTyDnMS2 "OMFG")

Reply:
![Heart attack](https://media.giphy.com/media/f7GXfUvpLXNTyDnMS2/giphy.gif)